activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1009,14 +1009,24 @@ def arel_columns(columns)
     end
 
     def reverse_sql_order(order_query)
-      order_query = ["#{quoted_table_name}.#{quoted_primary_key} ASC"] if order_query.empty?
+      if order_query.empty?
+        if primary_key
+          order_query = [table[primary_key].asc]
+        else
+          raise UnreversableOrderError, 
+            "Relation has no current order and table has no primary key to be used as default order" 
+        end
+      end
 
       order_query.flat_map do |o|
         case o
         when Arel::Nodes::Ordering
           o.reverse
         when String
-          o.to_s.split(',').map! do |s|
+          if o =~ /\([^()]*,[^()]*\)/
```

Comment:
Can we pull this out into a method that states what it's looking for, like `contains_functions_or_complex_sql_expressions_or_whatever_i_dont_thing_regexes_are_great_at_expressing_intent?(o)`